libraries/shared/watcher/event_watcher.go

```diff
@@ -97,16 +97,16 @@ func (watcher *EventWatcher) Execute(recheckHeaders constants.TransformerExecuti
 	}
 }
 
-func (watcher *EventWatcher) extractLogs(recheckHeaders constants.TransformerExecution, errs chan error) {
+func (watcher *EventWatcher) extractLogs(recheckHeaders constants.TransformerExecution, errs chan error) error {
 	call := func() error { return watcher.LogExtractor.ExtractLogs(recheckHeaders) }
-	watcher.withRetry(call, logs.ErrNoUncheckedHeaders, "extracting", errs)
+	return watcher.withRetry(call, logs.ErrNoUncheckedHeaders, "extracting", errs)
 }
 
-func (watcher *EventWatcher) delegateLogs(errs chan error) {
-	watcher.withRetry(watcher.LogDelegator.DelegateLogs, logs.ErrNoLogs, "delegating", errs)
+func (watcher *EventWatcher) delegateLogs(errs chan error) error {
+	return watcher.withRetry(watcher.LogDelegator.DelegateLogs, logs.ErrNoLogs, "delegating", errs)
```

Comment:
I don't think that this method and `extractLogs` actually need to return errors - since they're both being wrapped in goroutines, whenever the method finishes, the goroutine finishes. But would love another set of eyes on that assumption.

Reply:
👍 Agreed, think we can probably leave this unchanged - that's probably even preferable since calling in a goroutine prevents us from handling the error